### ISSUE TITLE
Pack with extra options fails for pyinstaller 

### ISSUE THREAD

**Issue by SolomidHero** [dashingsoft/pyarmor#741]
#740

**Comment by jondy** [comment]
Please first make sure PyInstaller could work with these options, as I have known, `--onefile` doesn't work in Pyinstaller 5.0

**Comment by SolomidHero** [comment]
Hi, @jondy!
Sorry I haven't written down all the command I used for pyarmor
All command was like this (`macos` platform)^
```bash
pyarmor pack --clean \
  -x ' --recursive' \
  -e ' --noconfirm --console --onedir --target-architecture x86_64 --add-data="main.yaml:./" --additional-hooks-dir hooks' \
  --keep -n Name main.py
```

As you can see I don't use `--onefile` option  (but `--onedir`) but I get error mentioned in #740 . 

Also, I can successfully build my project with providing generated .spec file
```bash
pyarmor pack -x ' --recursive'  -s main.spec main.py
```

**Comment by SolomidHero** [comment]
I can test packing with some versions of `pyinstaller<5.0` and write back if needed

**Comment by jondy** [comment]
This pr has been merged and release with new version 7.4.3
